compiler/src/org.graalvm.compiler.lir.aarch64/src/org/graalvm/compiler/lir/aarch64/AArch64AtomicMove.java

```diff
@@ -80,23 +85,35 @@ public void emitCode(CompilationResultBuilder crb, AArch64MacroAssembler masm) {
             Register address = asRegister(addressValue);
             Register result = asRegister(resultValue);
             Register newVal = asRegister(newValue);
+            if (masm.isFlagSet(AArch64.Flag.UseBarriersForVolatile) || useBarriers) {
+                masm.dmb(AArch64Assembler.BarrierKind.ANY_ANY);
+                emitCompareAndSwap(masm, size, address, result, newVal, false /* acquire */, false /* release */);
+                masm.dmb(AArch64Assembler.BarrierKind.ANY_ANY);
+            } else {
+                emitCompareAndSwap(masm, size, address, result, newVal, true /* acquire */, true /* release */);
+                if (!AArch64LIRFlagsVersioned.useLSE(masm.target.arch)) {
+                    // Temporarily add a fullbarrier after STLXR. This can be removed when LDAR is
+                    // used for java volatile read.
+                    masm.dmb(AArch64Assembler.BarrierKind.ANY_ANY);
+                }
```

Comment:
Temporarily insert a full barrier after each atomic operations, even though it is a non-conservative model. This is added here to resolve the renassiance banchmark hang issue (https://github.com/oracle/graal/issues/2045) when "STLXR" is used together with "LDR/DMB" for java volatile read. This issue can also be resolved once "LDAR/STLR" is used for java volatile access. So once the PR https://github.com/oracle/graal/pull/1772 is merged, the temporarily codes can be deleted.

Reply:
Yes that will do until the other PR gets pushed.

Reply:
That condition on "useLSE" doesn't make much sense to me. LSE instructions don't have full barrier semantics any more than LDAXR/STLXR instructions do.

Reply:
This temporary codes have been removed since PR 1772 has been merged. Thanks!